src/libsyntax/ext/expand.rs

```diff
@@ -1001,7 +1040,14 @@ impl<'a, 'b> Folder for InvocationCollector<'a, 'b> {
 
         // we'll expand attributes on expressions separately
         if !stmt.is_expr() {
-            let (attr, derives, stmt_) = self.classify_item(stmt);
+            let (attr, derives, stmt_) = if stmt.is_item() {
```

Comment:
I think I could flatten these two conditionals plus the `if let StmtKind::Mac(_)` to a single `match` if it's preferred.